api/src/main/java/jakarta/persistence/EntityAgent.java

```diff
@@ -0,0 +1,266 @@
+     *               with an assigned identifier
+     * @throws IllegalArgumentException if the entity has a null id
+     */
+    void upsert(Object entity);
+
+    /**
+     * Upsert every record in the given list. The records are
+     * upserted in the order in which they occur in the given list.
+     *
+     * @param entities The entities to be inserted or updated.
+     *
+     * @see #upsert(Object)
+     */
+    void upsertMultiple(List<?> entities);
+
+    /**
+     * Retrieve a record.
+     *
+     * @param entityClass The class of the entity to retrieve
+     * @param id The id of the entity to retrieve
+     *
+     * @return a detached entity instance
+     */
+    <T> T get(Class<T> entityClass, Object id);
+
+    /**
+     * Retrieve a record, obtaining the specified lock mode.
+     *
+     * @param entityClass The class of the entity to retrieve
+     * @param id The id of the entity to retrieve
+     * @param lockMode The lock mode to apply to the entity
+     *
+     * @return a detached entity instance
+     */
+    <T> T get(Class<T> entityClass, Object id, LockModeType lockMode);
+
+    /**
+     * Retrieve a record, fetching associations specified by the
+     * given {@link EntityGraph}.
+     *
+     * @param graph The {@link EntityGraph}
+     * @param graphSemantic a {@link GraphSemantic} specifying
+     *                      how the graph should be interpreted
+     * @param id The id of the entity to retrieve
+     *
+     * @return a detached entity instance
+     *
+     * @since 6.3
+     */
+    <T> T get(EntityGraph<T> graph, Object id);
+
+    /**
+     * Retrieve a record, fetching associations specified by the
+     * given {@linkplain EntityGraph load graph}, and obtaining
+     * the specified {@link LockModeType lock mode}.
+     *
+     * @param graph The {@linkplain EntityGraph load graph}
+     * @param id The id of the entity to retrieve
+     * @param lockMode The lock mode to apply to the entity
+     *
+     * @return a detached entity instance
+     */
+    <T> T get(EntityGraph<T> graph, Object id, LockModeType lockMode);
+
+    /**
+     * Retrieve multiple rows, returning entity instances in a
+     * list where the position of an instance in the list matches
+     * the position of its identifier in the given array, and the
+     * list contains a null value if there is no persistent
+     * instance matching a given identifier.
+     *
+     * @param entityClass The class of the entity to retrieve
+     * @param ids         The ids of the entities to retrieve
+     * @return an ordered list of detached entity instances, with
+     *         null elements representing missing entities
+     */
+    <T> List<T> getMultiple(Class<T> entityClass, List<?> ids);
+
+    /**
+     * Refresh the entity instance state from the database.
+     *
+     * @param entity The entity to be refreshed.
+     */
+    void refresh(Object entity);
```

Comment:
It might be nice to know if the entity was refreshed
```suggestion
     * @return {@code true} if the entity was found in the database,
     *             otherwise {@code false}.  
     */
    boolean refresh(Object entity);
```